### PR TITLE
Host state root

### PR DESCRIPTION
**Pull request by krl**

### CODE REVIEW

**Comment by krl** [comment]
@ZER0 sorry for removing your test for now, we need to discuss how to deal with this since i'm not sure if i understand exactly what you're trying to do with the externals.

File: canon_host/tests/root_test.rs [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+const APPLY_CONTRACT_TRANSACTION: u8 = 1;
+
+// queries
+const EXECUTE_CONTRACT_QUERY: u8 = 0;
+
+#[derive(Clone, Canon, Default)]
+struct TestState<S: Store> {
+    opcount: u64,
+    contracts: Vec<Remote<S>>,
+}
```
Comment (CPerezz) [inline canon_host/tests/root_test.rs:31]: I'm unsure how this is going to work once we integrate it. 

Does this mean that we will need to append to this structure all of the contracts? Is this a structure just created to generate delagate calls?

**Review by CPerezz — CHANGES_REQUESTED** [review]
Hey. I like a lot the solution overall, just have some questions that left in the code.

I only have one problem which is the fact that this PR appart from implementing inter-contract calls is removing the work done in #56  and #49 .

Could you rebase to master in order to avoid this removal?

**Comment by CPerezz** [comment]
Considering that this is going to be moved to `rusk_vm` I would either set it as draft so that the code is still reachable or if you have it on some other place, just close the PR and remove the branch.

Any suggestions @krl ?

**Pull request closed without merging**